Objective-C/CBLReplicator.mm

```diff
@@ -251,12 +251,17 @@ - (bool) _setupC4Replicator: (C4Error*)outErr {
 
     // Create a C4Replicator:
     CBL_LOCK(_config.database) {
+        [_config.database mustBeOpenLocked];
+        
         if (remoteURL || !otherDB)
             _repl = c4repl_new(_config.database.c4db, addr, dbName, params, outErr);
         else  {
 #ifdef COUCHBASE_ENTERPRISE
-            if (otherDB)
+            if (otherDB) {
+                [otherDB mustBeOpenLocked];
```

Comment:
This doesn't prevent the other thread to close the otherDB when creating the replicator. I think otherDB needs to be locked as well

Reply:
question, isn't Jay already call the mustBeOpenLocked from otherDB?

Reply:
after this line, it can get closed, before replicator is being started (in `c4repl_newLocal`)